README.md

```diff
@@ -95,6 +95,24 @@ var patches = [
 jsonpatch.apply( myobj, patches );
 // myobj == { firstName:"Joachim", lastName:"Wester", contactDetails:{ phoneNumbers[ {number:"555-123"} ] } };
 ```
+
+#### Applying patches on root:
+
+Applying patches on root is a little tricky, for that you can't replace your root object with an object of a different type in place (eg. can't replace `{item: "item 1"}` with `[{item: "item 1"}]`) on the root level, because the original document is an object `{}` and the new value is an array `[]`, nor can you do the opposite (`[] => {}`), obviously. See https://github.com/Starcounter-Jack/JSON-Patch/issues/147.
+
+However, you can use `applyPatch` method to over come this issue, because it returns the result object, and this allows it to change types.
```

Comment:
I think it would be great to document the usage with `array.reduce()` to apply multiple patches (that may modify the root).